src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -284,41 +284,80 @@ def makeSubscriptions(self, workflow):
-                        return workflow
-
-                    try:
-                        # In the message bellow we may want to put the list of datasets too
-                        msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
-                        self.logger.info(msg)
-                        ddmResult = self.ddm.makeRequest(ddmRequest)
-                    except Exception as ex:
-                        msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
-                        msg += "Error: {}".format(ex)
-                        self.logger.exception(msg)
-                        return workflow
+                        ddmReqList.append(ddmRequest)
+                except Exception as ex:
+                    msg = "Could not create DDMReq for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
+
+                try:
+                    # In the message bellow we may want to put the list of datasets too
+                    msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
+                    self.logger.info(msg)
+                    ddmResultList = self.ddm.makeAggRequests(ddmReqList, aggKey='item')
+                except Exception as ex:
+                    msg = "Could not make transfer subscription for Workflow: {}".format(workflow['RequestName'])
+                    msg += "Error: {}".format(ex)
+                    self.logger.exception(msg)
+                    return workflow
 
+                ddmStatusList = ['new', 'activated', 'completed', 'rejected', 'cancelled']
+                transferIDs = []
+                transferStatusList = []
+                for ddmResult in ddmResultList:
                     if 'data' in ddmResult.keys():
-                        self.docKeyUpdate(workflow,
-                                          transferStatus=deepcopy(ddmResult['data'][0]['status']),
-                                          transferIDs=deepcopy(ddmResult['data'][0]['request_id']))
-                        return workflow
-                    else:
-                        msg = "No data found in ddmResult for %s. Either dry run mode or " % workflow['RequestName']
-                        msg += "broken transfer submission to DDM. "
-                        msg += "ddmResult: \n%s" % pformat(ddmResult)
-                        self.logger.warning(msg)
+                        id = deepcopy(ddmResult['data'][0]['request_id'])
+                        status = deepcopy(ddmResult['data'][0]['status'])
+                        transferStatusList.append({'transferID': id,
+                                                   'status': status})
+                        transferIDs.append(id)
+
+                if transferStatusList and all(map(lambda x:
```

Comment:
replace map built-in by list comprehension (as reported in the jenkins results)

Reply:
BTW, you do not check the response from DDM, just that it returned any known status.

Reply:
The list comprehension is ready now. 
> you do not check the response from DDM

The DDM object always returns a response. Whether it will contain [data] depends on the run mode. If no Subscriptions are made for real the returned object is the DDMrequest itself (no [data] key here) and if the run mode is real then the actual result from the request is returned